client/concordclient/include/client/concordclient/event_update_queue.hpp

```diff
@@ -0,0 +1,132 @@
+// Concord
+//
+// Copyright (c) 2020-2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <condition_variable>
+#include <list>
+#include <memory>
+#include <mutex>
+
+#include "client/concordclient/event_update.hpp"
+
+namespace concord::client::concordclient {
+
+// An implementation of UpdateQueue should guarantee that ReleaseConsumers, Clear, Push, Pop, and TryPop are all
+// mutually thread safe.
+class UpdateQueue {
+ public:
+  // Destructor for UpdateQueue (which UpdateQueue implementations should
+  // override). Note the UpdateQueue interface does NOT require that
+  // implementations guarantee the desctructor be thread safe the UpdateQueue
```

Comment:
Nit: typo - `destructor`, and - `The UpdateQueue functions`, should be in a separate line I think.